parserator_web/views.py

```diff
@@ -13,12 +13,42 @@ class Home(TemplateView):
 class AddressParse(APIView):
     renderer_classes = [JSONRenderer]
 
+    errorMessage = "Unable to parse this value due to repeated labels."
+
     def get(self, request):
         # TODO: Flesh out this method to parse an address string using the
         # parse() method and return the parsed components to the frontend.
-        return Response({})
+
+        input_dict = request.GET  # the querydict -XM
+        input_string = input_dict.get("address")  # the string the user entered -XM
+        address_components = self.parse(input_string)[0]
+        address_type = self.parse(input_string)[1]
+
+        # The address type will be the errorMessage if an error was raised -XM
+        if address_type == self.errorMessage:
```

Comment:
this if/else block basically replicates the try/except block you have in the parse block. what about moving the try/except block up to here?

Reply:
True! I believe here I was worried about the exception getting raised in the parse function and the process being stopped there, forgetting that the exception would be caught higher up in the stack. So replacing this if/else with the try/except would absolutely be the move.